src/Uno.UI/UI/Xaml/Media/Brush.skia.cs

```diff
@@ -197,25 +197,28 @@ private static IDisposable AssignAndObserveRadialGradientBrush(RadialGradientBru
 			return disposables;
 		}
 
-		/// <summary>
-		/// Apply fallback colour for unimplemented <see cref="XamlCompositionBrushBase"/> types. For implemented types a more specific method
-		/// should be supplied.
-		/// </summary>
 		private static IDisposable AssignAndObserveXamlCompositionBrush(XamlCompositionBrushBase brush, Compositor compositor, BrushSetterHandler brushSetter)
 		{
 			var disposables = new CompositeDisposable();
 
-			var compositionBrush = compositor.CreateColorBrush(brush.FallbackColorWithOpacity);
+			brush.OnConnectedInternal();
+			var compositionBrush = brush.CompositionBrush ?? compositor.CreateColorBrush(brush.FallbackColorWithOpacity);
 
 			brush.RegisterDisposablePropertyChangedCallback(
 				XamlCompositionBrushBase.FallbackColorProperty,
-				(s, colorArg) => compositionBrush.Color = brush.FallbackColorWithOpacity
+				(s, colorArg) => { if (compositionBrush is CompositionColorBrush colorBrush) colorBrush.Color = brush.FallbackColorWithOpacity; }
```

Comment:
Looks like this statement is repeated multiple times, we could probably create a method like `Brush.TrySetColorFromBrush` or similar, which does the same compositionbrush test.